src/source.rs

```diff
@@ -903,21 +904,21 @@ impl SacnSourceInternal {
         if dst_ip.is_some() {
             self.socket
                 .send_to(&packet.pack_alloc().unwrap(), &dst_ip.unwrap().into())
-                .chain_err(|| "Failed to send data unicast on socket")?;
+                .with_context(|| "Failed to send data unicast on socket".to_string())?;
         } else {
             let dst;
 
             if self.addr.is_ipv6() {
                 dst = universe_to_ipv6_multicast_addr(universe)
-                    .chain_err(|| "Failed to convert universe to Ipv6 multicast address")?;
+                    .with_context(|| "Failed to convert universe to Ipv6 multicast address".to_string())?;
             } else {
                 dst = universe_to_ipv4_multicast_addr(universe)
-                    .chain_err(|| "Failed to convert universe to Ipv4 multicast address")?;
+                    .with_context(|| "Failed to convert universe to Ipv4 multicast address".to_string())?;
             }
 
             self.socket
                 .send_to(&packet.pack_alloc().unwrap(), &dst)
-                .chain_err(|| "Failed to send data multicast on socket")?;
+                .with_context(|| "Failed to send data multicast on socket".to_string())?;
```

Comment:
Use specific errors (potentially nested) to indicate these cases rather than using context